### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by RagnarR**
* Fix memory leak during `HttpURLConnection` in `doDrill()`
* Add new method for possibility set limit for requests.

### CODE REVIEW

File: url_driller/src/main/java/net/pubnative/URLDriller.java [hunk 1]
```diff
@@ -76,6 +77,180 @@ public void setUserAgent(String userAgent) {
...
+                    if(newUrl.startsWith("/")) {
+                        String protocol = urlObj.getProtocol();
+                        String host = urlObj.getHost();
+                        newUrl = protocol + "://" + host + newUrl;
+                    }
+                    invokeRedirect(newUrl);
+                    conn.disconnect();
+                    if (counter < mDrillSize) {
+                        doDrill(newUrl, counter + 1);
+                    }
```
Comment (anshumansworld) [inline url_driller/src/main/java/net/pubnative/URLDriller.java:160]: there should be else condition that will call `invokeFail(<>)`, no?

File: url_driller/src/main/java/net/pubnative/URLDriller.java [hunk 2]
```diff
@@ -76,6 +77,180 @@ public void setUserAgent(String userAgent) {
...
+     * This method will set user agent in request
+     *
+     * @param userAgent User-Agent string
+     */
+    public void setUserAgent(String userAgent) {
+
+        mUserAgent = userAgent;
+    }
+
+    /**
```
Comment (anshumansworld) [inline url_driller/src/main/java/net/pubnative/URLDriller.java]: I think, you missed to remove/comment unused code. It won't compile.

**Review by anshumansworld — CHANGES_REQUESTED** [review]

File: url_driller/src/main/java/net/pubnative/URLDriller.java (hunk 1 above)
Comment (RagnarR) [reply url_driller/src/main/java/net/pubnative/URLDriller.java:160]: make sense 👍

File: url_driller/src/main/java/net/pubnative/URLDriller.java (hunk 2 above)
Comment (RagnarR) [reply url_driller/src/main/java/net/pubnative/URLDriller.java]: Looks like was broken during creating PR. 👍

**Review by anshumansworld — APPROVED** [review]
👍

**Comment by RagnarR** [comment]
[(2) Refactor URLDriller to handle redirects](https://trello.com/c/NlMY9wUv/27-2-refactor-urldriller-to-handle-redirects)

**Pull request merged**